kb/development/mbedtls-coding-standards.md

```diff
@@ -128,6 +128,24 @@ Function parameters and local variables need no name spacing. They should use de
 
 By default all lengths and sizes are in bytes (or in number of elements, for arrays). If a name refers to a length or size in bits (as is often the case for key sizes) then the name must explicitly include `bit`, for example `mbedtls_pk_get_bitlen()` returns the size of the key in bits, while `mbedtls_pk_get_len()` returns the size in bytes. In addition, the documentation should always mention explicitly if key sizes are in bits or in bytes.
 
+Size refers to the capacity of the buffer, while length to the length of the contents. Most of the time these are interchangeable. A typical exception is when the output is written in a buffer, but the exact length is not known by the caller.
+
+In PSA code `length` and `size` in the name always implies bytes, when referring to number of elements, `count` is used.
+
+### Modules: `bignum_core`, `bignum_mod` and `bignum_mod_raw`
+
+Generic conventions:
+
+- `mbedtls_mpi_uint*` operands should be named by capital letters starting at the beginning of the alphabet (`A`, `B`, `C`, ...).
+- `mbedtls_mpi_uint` operands in turn should be named by lower case letters starting at the beginning of the alphabet (`a`, `b`, `c`)
+- For the result `X` or `x` should be used depending on the type.
+- One of the exceptions from this convention is where the naming of function parameters and local variables follows the literature (eg. Handbook of Applied Cryptography)
+
```

Comment:
While this is all well and good to represent `X := A + b * C`, it's less appropriate for things like `montmul` where we have things like `N` for the modulus, `mm` for `mpi_montg_init()` result input and `T` for temporary.

```
- `N` is generally used for the modulus.
- `T` is generally used for a temporary work area given to a function.
- Other specialised parameters (e.g. `mm`) might have their own standardised naming.
- ```
-

Reply:
That is why it is should and not must. Anyway, I added `N` and `T` to the guidelines.

(I am not particularly happy about the naming of the init result and I don't think it is a generic enough to be captured here.)